From: Lukasz Wasylyk <lukasz.wasylyk@redembedded.com>
Date: Mon, 21 Mar 2022 13:00:00 +0100
Subject: [PATCH] [ARRISEOS-41575] Extend timeout to 25 seconds

There is a risk that container can stop its work not completing previous request from slauncher.
Dbus timeout is set to be 18 secs. In the same time, once controller will notice stop of the container
it starts the process of killing the container. Before this change it was 10 sec of timeout.
Dbus interface is syncrhonous and confirmation of stop may not pass due to this 18 sec timeout.
This patch extends timeout for next 10 sec and next 5 , having checks of the status in the intervals:
10 sec, 10 sec ,5 sec. 25 sec in total. 

Index: git/Source/com/Communicator.cpp
===================================================================
--- git.orig/Source/com/Communicator.cpp
+++ git/Source/com/Communicator.cpp
@@ -180,12 +180,11 @@ namespace RPC {
                         _process.Kill(true);
                         nextinterval = 4000;
                     } else {
-                        ASSERT(false);
-                        nextinterval = 0;
+                        nextinterval = 10000;
                     }
                 } break;
                 case 2:
-                    _container->Stop(0);
+                    if(_process.Id() != 0) _container->Stop(0);
                     nextinterval = 5000;
                     break;
                 default:
